back/src/core/security/crypto.ts

```diff
@@ -1,24 +1,27 @@
-import * as crypto from "node:crypto"
-import { env } from "node:process"
-require("dotenv").config()
+import { CRYPTO_KEY_LENGTH, CRYPTO_SECRET } from "@env";
+import * as crypto from "node:crypto";
+import { Inject, Service } from "typedi";
 
-export const generateRandomSalt = (): string => {
-  return crypto.randomBytes(keyLength).toString("hex")
-}
-const keyLength = Number(env.CRYPTO_KEY_LENGTH!)
-const defaultSalt = env.CRYPTO_SECRET!
-
-export const generatePasswordWithSalt = (
-  value: string,
-  salt: string
-): string => {
-  if (!salt.length) {
-    throw new Error("Invalid salt")
+@Service()
+export class CryptoService {
+  constructor(
+    @Inject(CRYPTO_SECRET) private defaultSalt: string,
+    @Inject(CRYPTO_KEY_LENGTH) private keyLength: number
+  ) { }
+  
+  public generateRandomSalt(): string {
+    return crypto.randomBytes(this.keyLength |0).toString("hex")
```

Comment:
isso vai dar erro se lib pq ela nao aceita 0 bytes de tamanho. O ideial eh o servidor funcionar apenas com as envs, entao o ideal seria testar se a env existe antes de rodar o servidor, e ao longo do cdg, sempre considerar q o valor existe. Dessa forma, vc nao tem que ficar add essas condicoes td vez q for usar uma env. Faz sentido?

Reply:
Eu também achei estranho isso, mas é o operador bitwize. Aí se torna uma alternativa para converter para número. 
Como imaginei que fosse mais rápido que usando a classe Number resolvi adotar ele.
E infelizmente ou eu converto aí antes de usar ou o faço no constructor. 
Token<number> retorna string